src/batch_uploader.c

```diff
@@ -1058,7 +1058,8 @@ _key_put_submit_callback(as_error* ae, void* udata, as_event_loop* event_loop)
 			break;
 	}
 
-	if (as_aaf_uint64(&tracker->outstanding_calls, -1lu) == 0) {
+	if (as_aaf_uint64_rls(&tracker->outstanding_calls, -1lu) == 0) {
+		as_fence_seq();
```

Comment:
Could be made a `acq`.  We might want to add comments about why these are needed as well and in what case we can remove these memory barriers.  Like, if we switch the c11 I assume this will add redundant barriers.

Reply:
Changed to acquire fences. Left a comment above one of them.

Reply:
I think something more explicit like an inline `// remove after switching to c11` or something but I think it is fine